rosidl_gen/action_msgs.js

```diff
@@ -0,0 +1,213 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
```

Comment:
This files contains the required spec objects that .action files cannot provide. This information can only be obtained from the .idl files currently.

Reply:
For reference, [this](https://github.com/ros2/rosidl/blob/f0e95ee80d274e9948a052e3a509046f674d79f3/rosidl_adapter/rosidl_adapter/parser.py#L846) is what the .action file gives us, and [this](https://github.com/ros2/rosidl/blob/f0e95ee80d274e9948a052e3a509046f674d79f3/rosidl_parser/rosidl_parser/definition.py#L585) is what the .idl provides.

From the second link you can see how `send_goal_service`, `get_result_service`, and `feedback_message` are basically hardcoded.

Reply:
I basically came to the same conclusion that we'd need to use Python to generate the messages.

I think I read somewhere that ROS plans on making it easier for other languages to generate actions We might be able to make improvements at that time.